src/collection/AbstractCollection.php

```diff
@@ -29,7 +29,7 @@ public function clear(): void {
 	 * @internal
 	 */
 	public function rewind() {
-		return reset($this->array);
+		reset($this->array);
```

Comment:
both `reset()` and `next()` do have return values. Why you don't want to give them back any longer? Not that I'd say they should be used, but some `while ($item = $coll->next())` will benefit from it and probably generators, too. Does php has them? 😂

My idea why they should stop returning something is because of a used interface?

Reply:
Yep, collections implement `\Iterator` interface and `rewind` and `next` methods should return `void` (https://www.php.net/manual/en/class.iterator.php). Anyway, we can maintain the return values if it'll benefit.

Reply:
thanks for the explanation